src/components/modal/ModalDialog.js

```diff
@@ -58,20 +60,25 @@ class ModalDialog extends React.Component<Props> {
      * @return {ReactElement|null} - Returns the button, or null if the button shouldn't be rendered
      */
     renderCloseButton() {
-        const { closeButtonProps, onRequestClose, intl } = this.props;
+        const { closeButtonProps, closeButtonText, onRequestClose, intl } = this.props;
         const { formatMessage } = intl;
         if (!onRequestClose) {
             return null;
         }
 
-        return (
+        const sharedProps = {
+            ...closeButtonProps,
+            'aria-label': formatMessage(messages.closeModalText),
+            onClick: this.onCloseButtonClick,
+        };
+
+        return closeButtonText ? (
+            <Button className="bdl-ModalCloseButton--with-custom-text" {...sharedProps}>
+                {closeButtonText}
+            </Button>
+        ) : (
```

Comment:
1 thing to test, if button text is too long or of modal title is too long or both are too long...overflows. Maybe ellipsis might be needed